src/components/radio/radio.ts

```diff
@@ -0,0 +1,140 @@
+import { html, LitElement } from 'lit';
+import { property, query } from 'lit/decorators.js';
+import { styles } from './radio.material.css';
+import { EventEmitterMixin } from '../common/mixins/event-emitter.js';
+import { Constructor } from '../common/mixins/constructor.js';
+import { watch } from '../common/decorators';
+import { ifDefined } from 'lit/directives/if-defined.js';
+import { live } from 'lit/directives/live.js';
+
+let nextId = 0;
+
+export interface IgcRadioEventMap {
+  igcChange: CustomEvent<void>;
+  igcFocus: CustomEvent<void>;
+  igcBlur: CustomEvent<void>;
+}
+
+export class IgcRadioComponent extends EventEmitterMixin<
+  IgcRadioEventMap,
+  Constructor<LitElement>
+>(LitElement) {
+  static styles = styles;
+
+  private inputId = `radio-${nextId++}`;
+  private labelId = `radio-label-${this.inputId}`;
+
+  @query('input[type="radio"]')
+  input!: HTMLInputElement;
+
+  @property()
+  name!: string;
+
+  @property()
+  value!: string;
+
+  @property({ type: Boolean, reflect: true })
+  checked = false;
+
+  @property({ type: Boolean, reflect: true })
+  disabled = false;
+
+  @property({ type: Boolean, reflect: true })
+  invalid = false;
+
+  @property({ reflect: true, attribute: 'label-position' })
+  labelPosition: 'before' | 'after' = 'after';
+
+  @property({ reflect: true, attribute: 'aria-labelledby' })
+  ariaLabelledby!: string;
+
+  click() {
+    this.input.click();
+  }
+
+  focus(options?: FocusOptions) {
+    this.input.focus(options);
+  }
+
+  blur() {
+    this.input.blur();
+  }
+
+  reportValidity() {
+    this.input.reportValidity();
+  }
+
+  setCustomValidity(message: string) {
+    this.input.setCustomValidity(message);
+    this.invalid = !this.input.checkValidity();
+  }
+
+  handleMouseDown(event: MouseEvent) {
+    event.preventDefault();
+    this.input.focus();
+  }
+
+  handleClick() {
+    this.checked = true;
+  }
+
+  handleBlur() {
+    this.emitEvent('igcBlur');
+  }
+
+  handleFocus() {
+    this.emitEvent('igcFocus');
+  }
+
+  @watch('checked', { waitUntilFirstUpdate: true })
+  handleChange() {
+    if (this.checked) {
+      this.getSiblings().forEach((radio) => {
+        radio.checked = false;
+      });
+      this.input.focus();
+      this.emitEvent('igcChange');
+    }
+  }
+
+  getSiblings() {
+    const group = this.closest('igc-radio-group');
+    if (!group) return [];
+
+    return Array.from<IgcRadioComponent>(
+      group.querySelectorAll('igc-radio')
+    ).filter((radio) => radio.name === this.name && radio !== this);
+  }
+
+  render() {
+    return html`
+      <label
+        part="base"
+        for="${this.inputId}"
+        @mousedown="${this.handleMouseDown}"
+      >
+        <input
+          id="${this.inputId}"
+          type="radio"
+          name="${ifDefined(this.name)}"
+          value="${ifDefined(this.value)}"
+          .disabled="${this.disabled}"
+          .checked="${live(this.checked)}"
+          tabindex=${this.checked ? 0 : -1}
```

Comment:
If there is no checked radio initially, we should make the first radio in a group focusable.

Reply:
It is better now but there is one minor difference with the standard radio behavior. If no radio is selected and you focus the first one then press arrow down, the second radio should be selected (in our implementation the first one is selected).

Reply:
This behavior just adds unnecessary complexity to the implementation, imo. If you really wanna have it that way - fine, but I think we can do without it.

Reply:
I am ok to leave it like this and fix in the future if a customer complains.